build_tools/cmake/ctest_all.sh

```diff
@@ -168,7 +168,10 @@ if (( ${#excluded_tests[@]} )); then
   ctest_args+=("--exclude-regex ${excluded_tests_regex}")
 fi
 
-echo "*************** Running CTest ***************"
-
 set -x
+
+echo "*************** Running CTest ***************"
 ctest ${ctest_args[@]}
+
+echo "**************** llvm-external-projects tests ***************"
+cmake --build . --target check-iree-dialects -- -k 0
```

Comment:
This is failing with

```
+ echo '**************** llvm-external-projects tests ***************'
+ cmake --build . --target check-iree-dialects -- -k 0
Error: could not load cache
**************** llvm-external-projects tests ***************
```
(logs here: https://github.com/openxla/iree/actions/runs/6826930880/job/18568177838?pr=15530#step:6:3065)

Probably a result of the split build -> test jobs not including all the files that `cmake --build` expects:
* https://github.com/openxla/iree/blob/main/.github/workflows/build_all.yml
* https://github.com/openxla/iree/blob/9fc8f821c958842847638cdae5a31b1dd2a169a8/.github/workflows/ci.yml#L299-L302

Reply:
Actually, this may be a case where the command should be run from the build directory, not the source directory. There may still be issues with cross compilation and other weird builds though.